app/src/test/java/io/gnosis/safe/ui/safe/settings/SafeSettingsViewModelTest.kt

```diff
@@ -26,36 +28,41 @@ class SafeSettingsViewModelTest {
 
     private lateinit var safeSettingsViewModel: SafeSettingsViewModel
 
-    @Before
-    fun setup() {
-        safeSettingsViewModel = SafeSettingsViewModel(safeRepository, tracker, appDispatchers)
-    }
 
     @Test
     fun `removeSafe - should remove safe`() = runBlockingTest {
 
         coEvery { safeRepository.getActiveSafe() } returnsMany listOf(SAFE_1, null)
+        coEvery { safeRepository.activeSafeFlow() } returns flow {
+            emit(SAFE_1)
+            emit(null)
+        }
+            .conflate()
```

Comment:
I am not sure you should be calling `conflate` there. The conflation of items should happen in the viewModel when it's ready to receive the flow

Reply:
this will change nothing in this test, as there is only one subscriber to the state

Reply:
https://kotlinlang.org/docs/reference/coroutines/flow.html

The part about conflations says: 

Conflation
When a flow represents partial results of the operation or operation status updates, it may not be necessary to process each value, but instead, only most recent ones. In this case, the conflate operator can be used to skip intermediate values when a collector is too slow to process them. Building on the previous example:

Reply:
That may be the reason you are losing some events in between.

Reply:
this was my intention behind using it.